apps/music/js/music.js

```diff
@@ -160,88 +160,116 @@ var App = (function() {
-    LazyLoader.load('js/metadata/album_art_cache.js', function() {
-      function showListView() {
-        var option = TabBar.option;
-        var info = {
-          key: 'metadata.' + option,
-          range: null,
-          direction: (option === 'title') ? 'next' : 'nextunique',
-          option: option
-        };
-
-        ListView.activate(info);
+  function dbEnumerable(callback) {
+    // If we've been upgrading, hide that now
+    if (app.currentOverlay === 'upgrade') {
+      app.showOverlay(null);
+    }
+
+    // Display music that we already know about
+    refreshViews(function() {
+      // Tell performance monitors that the content is displayed and is
+      // ready to interact with. We won't send the final moz-app-loaded
+      // event until we're completely stable and have finished scanning.
+      //
+      // XXX: Maybe we could emit these events earlier, when we've just
+      // finished the "above the fold" content. That's hard to do on
+      // arbitrary screen resolutions, though.
+      window.performance.mark('visuallyLoaded');
+      window.dispatchEvent(new CustomEvent('moz-app-visually-complete'));
+      window.performance.mark('contentInteractive');
+      window.dispatchEvent(new CustomEvent('moz-content-interactive'));
+
+      // For performance optimization, we disable the font-fit logic in
+      // gaia-header to speed up the startup times, and here we have to
+      // remove the no-font-fit attribute to trigger the font-fit logic.
+      TitleBar.view.removeAttribute('no-font-fit');
+
+      // Hide the spinner once we've displayed the initial screen
+      document.getElementById('spinner-overlay').classList.add('hidden');
+
+      // Only init the communication when music is not in picker mode.
+      if (document.URL.indexOf('#pick') === -1) {
+        // We need to wait to init the music comms until the UI is fully loaded
+        // because the init of music comms could slow down the startup time.
+        MusicComms.init();
       }
-      // If it's in picking mode, we will just enumerate all the songs. We don't
-      // need to enumerate data for TilesView because the mix page is not needed
-      // in picker mode.
-      if (app.pendingPick) {
-        showListView();
-        if (callback) {
-          callback();
-        }
-        return;
+
+      if (callback) {
+        callback();
+      }
+    });
+  }
+
+  function dbReady(refresh, callback) {
+    // Hide the nocard or pluggedin overlay if it is displayed
+    if (app.currentOverlay === 'nocard' || app.currentOverlay === 'pluggedin') {
+      app.showOverlay(null);
+    }
+
+    if (refresh) {
+      refreshViews(callback);
+    } else if (callback) {
+      callback();
+    }
+  }
+
+  // This tracks if we've automatically hidden the search box on startup yet (it
+  // should only be done once!)
+  var hidSearchBox = false;
+
+  function refreshViews(callback) {
```

Comment:
Actually, I have not reviewed this code... I just gave feedback on the 'enumerable' parts of it.  I'm assuming that most of this stuff is under review by Dominic and is here because the two patches are interdependent.  So I'm going to skip this stuff and just go looking for the album art related code.